src/tracer/MOM_offline_control.F90

```diff
@@ -0,0 +1,619 @@
+  use MOM_time_manager,     only : time_type
+  use MOM_domains,          only : pass_var, pass_vector, To_All
+  use MOM_error_handler,    only : callTree_enter, callTree_leave, MOM_error, FATAL, WARNING, is_root_pe
+  use MOM_grid,             only : ocean_grid_type
+  use MOM_verticalGrid,     only : verticalGrid_type
+  use MOM_io,               only : read_data
+  use MOM_file_parser,      only : get_param, log_version, param_file_type
+  use MOM_diag_mediator,    only : diag_ctrl, register_diag_field
+  use mpp_domains_mod,      only : CENTER, CORNER, NORTH, EAST
+  use MOM_variables,        only : vertvisc_type
+  use MOM_forcing_type,     only : forcing
+  use MOM_shortwave_abs,    only : optics_type
+  use MOM_diag_mediator,    only : post_data
+  use MOM_forcing_type,     only : forcing
+  use MOM_diabatic_driver,  only : diabatic_CS
+
+  implicit none
+
+#include <MOM_memory.h>
+
+  type, public :: offline_transport_CS
+
+    !> Variables related to reading in fields from online run
+    integer :: start_index  ! Timelevel to start
+    integer :: numtime      ! How many timelevels in the input fields
+    integer :: &            ! Index of each of the variables to be read in
+      ridx_sum = -1, &      ! Separate indices for each variabile if they are
+      ridx_snap = -1        ! setoff from each other in time
+    character(len=200) :: offlinedir  ! Directory where offline fields are stored
+    character(len=200) :: & !         ! Names of input files
+      snap_file,  &
+      sum_file
+    logical :: fields_are_offset ! True if the time-averaged fields and snapshot fields are
+                                 ! offset by one time level
+    
+    !> Variables controlling some of the numerical considerations of offline transport
+    integer           ::  num_off_iter
+    real              ::  dt_offline ! Timestep used for offline tracers
+    real              ::  max_off_cfl=0.5 ! Hardcoded for now, only used in non-ALE mode
+    real              ::  evap_CFL_limit, minimum_forcing_depth
+
+    !> Diagnostic manager IDs for use in the online model of additional fields necessary
+    !> for offline tracer modeling
+    integer :: &
+      id_uhtr_preadv = -1, &
+      id_vhtr_preadv = -1, &
+    !> Diagnostic manager IDs for some fields that may be of interest when doing offline transport  
+      id_uhr = -1, &
+      id_vhr = -1, &
+      id_ear = -1, &
+      id_ebr = -1, &
+      id_hr = -1
+
+  end type offline_transport_CS
+
+#include "MOM_memory.h"
+#include "version_variable.h"
+  public offline_transport_init
+  public transport_by_files
+  public register_diags_offline_transport
+  public update_h_horizontal_flux
+  public update_h_vertical_flux
+  public limit_mass_flux_3d
+
+contains
+
+  subroutine transport_by_files(G, GV, CS, h_end, eatr, ebtr, uhtr, vhtr, khdt_x, khdt_y, &
+      temp, salt, fluxes, do_ale_in)
+  !> Controls the reading in 3d mass fluxes, diffusive fluxes, and other fields stored
```

Comment:
Should be either `!<` but would prefer you place the comment above. The direction of the angle bracket indicates to doxygen to what the comment belongs. Current this subroutine description is associated with `G`.